sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -295,13 +298,25 @@ private[hive] object HadoopTableReader extends HiveInspectors with Logging {
   def initializeLocalJobConfFunc(path: String, tableDesc: TableDesc)(jobConf: JobConf) {
     FileInputFormat.setInputPaths(jobConf, Seq[Path](new Path(path)): _*)
     if (tableDesc != null) {
-      PlanUtils.configureInputJobPropertiesForStorageHandler(tableDesc)
+      configureJobPropertiesForStorageHandler(tableDesc, jobConf)
       Utilities.copyTableJobPropertiesToConf(tableDesc, jobConf)
     }
     val bufferSize = System.getProperty("spark.buffer.size", "65536")
     jobConf.set("io.file.buffer.size", bufferSize)
   }
 
+  private def configureJobPropertiesForStorageHandler(tableDesc: TableDesc, jobConf: JobConf) {
```

Comment:
can you add some comment explaining what's happening, and why this is done this way?